In [1]:
!git clone https://github.com/bert-nmt/bert-nmt

Cloning into 'bert-nmt'...
remote: Enumerating objects: 299, done.
remote: Counting objects: 100% (53/53), done.
remote: Compressing objects: 100% (31/31), done.
remote: Total 299 (delta 29), reused 22 (delta 22), pack-reused 246
Receiving objects: 100% (299/299), 2.59 MiB | 13.77 MiB/s, done.
Resolving deltas: 100% (59/59), done.


In [2]:
%cd bert-nmt/

/content/bert-nmt


In [3]:
!pip install --editable .

Obtaining file:///content/bert-nmt
     |████████████████████████████████| 90 kB 4.0 MB/s 
     |████████████████████████████████| 132 kB 32.2 MB/s 
     |████████████████████████████████| 79 kB 8.3 MB/s 
     |████████████████████████████████| 8.7 MB 9.1 MB/s 
     |████████████████████████████████| 138 kB 47.9 MB/s 
     |████████████████████████████████| 127 kB 50.1 MB/s 
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.24.3
    Uninstalling urllib3-1.24.3:
      Successfully uninstalled urllib3-1.24.3
  Running setup.py develop for fairseq
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.


In [4]:
%cd /content/bert-nmt/examples
!git clone http://github.com/moses-smt/mosesdecoder

/content/bert-nmt/examples
Cloning into 'mosesdecoder'...
remote: Enumerating objects: 148090, done.
remote: Counting objects: 100% (518/518), done.
remote: Compressing objects: 100% (223/223), done.
remote: Total 148090 (delta 319), reused 443 (delta 292), pack-reused 147572
Receiving objects: 100% (148090/148090), 129.87 MiB | 16.90 MiB/s, done.
Resolving deltas: 100% (114345/114345), done.


In [6]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
%cd /content/
!git clone https://github.com/fajri91/minangNLP.git

/content
Cloning into 'minangNLP'...
remote: Enumerating objects: 91, done.
remote: Counting objects: 100% (91/91), done.
remote: Compressing objects: 100% (83/83), done.
remote: Total 91 (delta 26), reused 47 (delta 6), pack-reused 0
Unpacking objects: 100% (91/91), done.


In [7]:
%cd /content/bert-nmt/examples/translation/

/content/bert-nmt/examples/translation


In [9]:
!chmod 777 ./makedataforbert.sh

Rerun with both languages

In [11]:
!./makedataforbert.sh jav

src lng jav
Detokenizer Version $Revision: 4134 $
Language: jav
Detokenizer Version $Revision: 4134 $
Language: jav
Detokenizer Version $Revision: 4134 $
Language: jav


In [ ]:
while True:
  continue

KeyboardInterrupt: ignored

In [ ]:
% cd /content/bert-nmt/

/content/bert-nmt


In fairseq/modules/multihead_attention.py line 132, change to:
q = q * self.scaling

In fairseq/models/transformer.py line 339, change to:
        bert_encoder_out, _ =  self.bert_encoder(bert_input, output_all_encoded_layers=True, attention_mask= 1. - bert_encoder_padding_mask.type(torch.LongTensor).cuda())


In [17]:
!chmod 777 ./something.sh

In [24]:
!./something.sh

Thu Apr 21 02:00:56 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla P100-PCIE...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   65C    P0    31W / 250W |      0MiB / 16280MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

In [ ]:
!ls

bert		      LICENSE			 train.bert.min-ind.min.bin
build		      PATENTS			 train.bert.min-ind.min.idx
checkpoints	      preprocess.py		 train.ind
CODE_OF_CONDUCT.md    README.md			 train.min
CONTRIBUTING.md       score.py			 train.min-ind.ind.bin
dict.ind.txt	      scripts			 train.min-ind.ind.idx
dict.min.txt	      setup.py			 train.min-ind.min.bin
docs		      something.sh		 train.min-ind.min.idx
eval_lm.py	      test.bert.ind		 train.py
examples	      test.bert.min		 valid.bert.ind
fairseq		      test.bert.min-ind.min.bin  valid.bert.min
fairseq_cli	      test.bert.min-ind.min.idx  valid.bert.min-ind.min.bin
fairseq.egg-info      test.ind			 valid.bert.min-ind.min.idx
fairseq.gif	      test.min			 valid.ind
fairseq_logo.png      test.min-ind.ind.bin	 valid.min
generate.py	      test.min-ind.ind.idx	 valid.min-ind.ind.bin
generator.py	      test.min-ind.min.bin	 valid.min-ind.ind.idx
hubconf.py	      test.min-ind.min.idx	 valid.min-ind.min.bin
interactive.py	      tests			 valid.min-in

In [25]:
while True:
  continue

KeyboardInterrupt: ignored

In [ ]:
! fairseq-generate /content/bert-nmt/  --path "/content/bert-nmt/checkpoints/iwed_minind0.5/checkpoint_best.pt" --batch-size 128 --beam 5 --remove-bpe --lenpen 0.5 --gen-subset test > wmt16_gen.txt

Traceback (most recent call last):
  File "/usr/local/bin/fairseq-generate", line 33, in <module>
    sys.exit(load_entry_point('fairseq', 'console_scripts', 'fairseq-generate')())
  File "/content/bert-nmt/fairseq_cli/generate.py", line 191, in cli_main
    main(args)
  File "/content/bert-nmt/fairseq_cli/generate.py", line 52, in main
    geargs=args,
  File "/content/bert-nmt/fairseq/checkpoint_utils.py", line 161, in load_model_ensemble
    state = load_checkpoint_to_cpu(filename)
  File "/content/bert-nmt/fairseq/checkpoint_utils.py", line 142, in load_checkpoint_to_cpu
    path, map_location=lambda s, l: default_restore_location(s, 'cpu'),
  File "/usr/local/lib/python3.7/dist-packages/torch/serialization.py", line 600, in load
    with _open_zipfile_reader(opened_file) as opened_zipfile:
  File "/usr/local/lib/python3.7/dist-packages/torch/serialization.py", line 242, in __init__
    super(_open_zipfile_reader, self).__init__(torch._C.PyTorchFileReader(name_or_buffer))
RuntimeEr

In [ ]:
! python interactive.py /content/bert-nmt/ --path /content/bert-nmt/checkpoints/iwed_minind0.5/checkpoint_best.pt -s min -t ind --bert-model-name bert-base-uncased --beam 5 --remove-bpe

Namespace(beam=5, bert_first=True, bert_gates=[1, 1, 1, 1, 1, 1], bert_model_name='bert-base-uncased', bert_output_layer=-1, bert_ratio=1.0, buffer_size=1, change_ratio=False, cpu=False, criterion='cross_entropy', data='/content/bert-nmt/', dataset_impl='cached', decoder_no_bert=False, diverse_beam_groups=-1, diverse_beam_strength=0.5, encoder_bert_dropout=False, encoder_bert_dropout_ratio=0.25, encoder_bert_mixup=False, encoder_ratio=1.0, finetune_bert=False, force_anneal=None, fp16=False, fp16_init_scale=128, fp16_scale_tolerance=0.0, fp16_scale_window=None, gen_subset='test', input='-', lazy_load=False, left_pad_source='True', left_pad_target='False', lenpen=1, log_format=None, log_interval=1000, lr_scheduler='fixed', lr_shrink=0.1, mask_cls_sep=False, match_source_len=False, max_len_a=0, max_len_b=200, max_sentences=1, max_source_positions=1024, max_target_positions=1024, max_tokens=None, memory_efficient_fp16=False, min_len=1, min_loss_scale=0.0001, model_overrides='{}', momentum=

In [ ]:
import torch
torch.cuda.empty_cache()

In [ ]:
import gc
gc.collect()

329

In [16]:
%cd /content/bert-nmt

/content/bert-nmt


In [ ]:
import torch
!pip install 

1.10.0+cu111


In [23]:
! python generate.py /content/bert-nmt/ \
    --path /content/bert-nmt/checkpoints/iwed_indjav0.5/checkpoint_best.pt \
    --batch-size 128 --remove-bpe --bert-model-name bert-base-uncased --lenpen 0.5 --gen-subset test

Streaming output truncated to the last 5000 lines.
S-395	Lalu firman Tuhan <unk> di seluruh daerah itu.
T-395	Pangandikané Pangéran banjur kawartakaké ing saindenging tanah kono.
H-395	-2.1748945713043213	Lajeng dipun aniaya déning Pangéran wonten ing tanah kono kabèh.
P-395	-2.0599 -0.3190 -0.4233 -0.0242 -0.4960 -0.3900 -0.0470 -0.5016 -2.0727 -0.8513 -0.0283
S-1060	Kasih karunia Tuhan Yesus Kristus menyertai roh kamu!
T-1060	Sih-rahmaté Gusti Yésus Kristus nunggila ing <<unk>>
H-1060	-0.7969021201133728	Sih-rahmaté Gusti Yésus Kristus nunggila ing rohmu , Sadulur-sadulur! Amin.
P-1060	-0.8073 -0.0124 -0.0217 -0.0540 -0.6021 -0.1595 -0.1420 -0.1442 -0.5012 -0.1424 -0.0563
S-125	Tunjukkanlah belas kasihan kepada mereka yang ragu-ragu ,
T-125	Wong-wong kang atiné isih mangu-mangu padha <<unk>>
H-125	-2.530932903289795	Padha ngedohana sarupaning panggawé becik.
P-125	-1.3661 -2.3413 -0.2073 -2.1282 -0.1178 -0.0388
S-1140	Yesus dan murid-murid-Nya diundang juga ke perkawinan 

In [ ]:
!git clone https://github.com/OpenNMT/OpenNMT-py.git
%cd OpenNMT-py
!python setup.py install

Cloning into 'OpenNMT-py'...
remote: Enumerating objects: 17675, done.
remote: Counting objects: 100% (631/631), done.
remote: Compressing objects: 100% (313/313), done.
^C
[Errno 2] No such file or directory: 'OpenNMT-py'
/content/bert-nmt
running install
running bdist_egg
running egg_info
writing fairseq.egg-info/PKG-INFO
writing dependency_links to fairseq.egg-info/dependency_links.txt
writing entry points to fairseq.egg-info/entry_points.txt
writing requirements to fairseq.egg-info/requires.txt
writing top-level names to fairseq.egg-info/top_level.txt
adding license file 'LICENSE'
writing manifest file 'fairseq.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build/lib.linux-x86_64-3.7/bert
copying bert/__init__.py -> build/lib.linux-x86_64-3.7/bert
copying bert/modeling.py -> build/lib.linux-x86_64-3.7/bert
copying bert/tokenization.py -> build/lib.linux-x86_64-3.7/bert
copying bert/file_utils.py -> build/l

In [12]:
%cd /content/bert-nmt/

/content/bert-nmt


In [15]:
! python preprocess.py --source-lang ind --target-lang jav --trainpref train --validpref valid --testpref test --destdir /content/bert-nmt --joined-dictionary --bert-model-name bert-base-cased

Namespace(alignfile=None, bert_model_name='bert-base-cased', cpu=False, criterion='cross_entropy', dataset_impl='cached', destdir='/content/bert-nmt', fp16=False, fp16_init_scale=128, fp16_scale_tolerance=0.0, fp16_scale_window=None, joined_dictionary=True, log_format=None, log_interval=1000, lr_scheduler='fixed', memory_efficient_fp16=False, min_loss_scale=0.0001, no_progress_bar=False, nwordssrc=-1, nwordstgt=-1, only_source=False, optimizer='nag', padding_factor=8, seed=1, source_lang='ind', srcdict=None, target_lang='jav', task='translation', tbmf_wrapper=False, tensorboard_logdir='', testpref='test', tgtdict=None, threshold_loss_scale=None, thresholdsrc=0, thresholdtgt=0, trainpref='train', user_dir=None, validpref='valid', workers=1)
| [ind] Dictionary: 23223 types
| [ind] train.ind: 5968 sents, 137704 tokens, 0.0% replaced by <unk>
| [ind] Dictionary: 23223 types
| [ind] valid.ind: 797 sents, 18522 tokens, 3.74% replaced by <unk>
| [ind] Dictionary: 23223 types
| [ind] test.ind:

In [14]:
!cp /content/bert-nmt/examples/translation/*.jav /content/bert-nmt/